File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numMap = new Dictionary<int, int>(); // A mapping to store numbers and their indices
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i]; // Find the required number to reach the target
            if (numMap.ContainsKey(complement)) {
                return new int[] { numMap[complement], i }; // Return indices of the complement and current number
            }
            numMap[nums[i]] = i; 
        }
        return new int[] {}; 
    }
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mp=new HashMap();

        for(int i=0;i<nums.length;i++){
            mp.put(nums[i], i);
        } 

        for(int i=0;i<nums.length;i++){
            int c=target - nums[i];
            if(mp.containsKey(c)){
                int index=mp.get(c);
                if(index==i) continue;
                return new int[] {i, index};
            }
        }
        
        return new int[] {-1, -1};

    }
}